### PR TITLE
protocol-change: sets event in case of failure

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
None

Describe changes:
- Sets an event if protocol change fails (ie if there is already protocol change going on)

Another way to do this would be to set a packet's app-layer event, but `AppLayerRequestProtocolChange` does not have access to the current `Packet` and its caller `FTPParseResponse` does not have it neither...
So, this PR implements different events for the different app-layer protocols.
FTP and PGSQL do not have events yet, and are not part of this...

### CODE REVIEW

**Comment by catenacyber** [comment]
Victor, you requested this event, so assigning this draft to you ;-)

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on tlpw1_files_sha256.
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk | | |
| .flow.memuse | 530312768 | 498197568 | 106.45% | 

Pipeline 7251

File: rules/http-events.rules [hunk 1]
```diff
@@ -87,4 +87,6 @@ alert http any any -> any any (msg:"SURICATA HTTP invalid Range header value"; f
 
 alert http any any -> any any (msg:"SURICATA HTTP file name too long"; flow:established; app-layer-event:http.file_name_too_long; flowint:http.anomaly.count,+,1; classtype:protocol-command-decode; sid:2221052; rev:1;)
 
-# next sid 2221053
+alert http any any -> any any (msg:"SURICATA HTTP failed protocol change"; flow:established; app-layer-event:http.failed_protocol_change; flowint:http.anomaly.count,+,1; classtype:protocol-command-decode; sid:2221053; rev:1;)
```
Comment (victorjulien) [inline rules/http-events.rules:90]: is there any case where this isn't about a HTTPS upgrade?

File: rules/smtp-events.rules [hunk 2]
```diff
@@ -30,4 +30,5 @@ alert smtp any any -> any any (msg:"SURICATA SMTP Mime boundary length exceeded"
 alert smtp any any -> any any (msg:"SURICATA SMTP duplicate fields"; flow:established,to_server; app-layer-event:smtp.duplicate_fields; flowint:smtp.anomaly.count,+,1; classtype:protocol-command-decode; sid:2220018; rev:1;)
 alert smtp any any -> any any (msg:"SURICATA SMTP unparsable content"; flow:established,to_server; app-layer-event:smtp.unparsable_content; flowint:smtp.anomaly.count,+,1; classtype:protocol-command-decode; sid:2220019; rev:1;)
 alert smtp any any -> any any (msg:"SURICATA SMTP filename truncated"; flow:established,to_server; app-layer-event:smtp.mime_long_filename; flowint:smtp.anomaly.count,+,1; classtype:protocol-command-decode; sid:2220020; rev:1;)
-# next sid 2220021
+alert smtp any any -> any any (msg:"SURICATA SMTP failed protocol change"; flow:established; app-layer-event:smtp.failed_protocol_change; flowint:smtp.anomaly.count,+,1; classtype:protocol-command-decode; sid:2220021; rev:1;)
```
Comment (victorjulien) [inline rules/smtp-events.rules:33]: this is always about STARTTLS, right?

**Comment by victorjulien** [comment]
Are there pcaps to test this scenario? If not can you craft them?

File: rules/http-events.rules (hunk 1 above)
Comment (catenacyber) [reply rules/http-events.rules:90]: HTTP1->HTTP2

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7745 with S-V test linked ;-)

**Pull request closed without merging**